modules/bene_core/src/Plugin/Block/SubNavigationBlock.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Drupal\bene_core\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Link;
+
+/**
+ * Provides a 'SubNavigationBlock' block.
+ *
+ * @Block(
+ *  id = "bene_sub_navigation_block",
+ *  admin_label = @Translation("Bene sub navigation"),
+ *  category = @Translation("Bene")
+ * )
+ */
+class SubNavigationBlock extends BlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [] + parent::defaultConfiguration();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $form = [];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockSubmit($form, FormStateInterface $form_state) {
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    // Disable block cache as page title will change per-page.
+    $build = [
+      '#cache' => [
+        'max-age' => 0,
+      ],
+    ];
+
+    // Add breadcrumbs excluding current page.
+    $breadcrumbs = $this->getBreadcrumbs();
+    $breadcrumb_links = [];
+    /** @var Link $breadcrumb */
+    foreach ($breadcrumbs as $breadcrumb) {
+      $breadcrumb_links[] = Link::fromTextAndUrl($breadcrumb->getText(), $breadcrumb->getUrl());
+    }
+
+    $build['breadcrumbs'] = [
+      '#theme' => 'item_list',
+      '#wrapper_attributes' => [
+        'class' => [
+          'bene-subnav-breadcrumbs',
+        ],
+      ],
+      '#items' => $breadcrumb_links,
+    ];
+
+    // Add current page title.
+    $page_title = $this->getPageTitle();
+    // Account for the page title being returned as either a string or render
+    // array.
+    if (is_array($page_title)) {
+      $build['page_title'] = $page_title;
+    }
+    else {
+      $build['page_title'] = [
+        '#type' => 'markup',
+        '#markup' => $this->getPageTitle(),
+        '#prefix' => '<div class="bene-subnav-page-title">',
+        '#suffix' => '</div>',
+      ];
+    }
+
+    $page_child_links = [];
+
+    $page_children = $this->getChildMenuLinks();
+    if (!empty($page_children)) {
+      foreach ($page_children as $page_child) {
+        $page_child_links[] = Link::createFromRoute($page_child['title'], $page_child['route_name'], $page_child['route_parameters']);
+      }
+
+      $build['page_children'] = [
+        '#theme' => 'item_list',
+        '#wrapper_attributes' => [
+          'class' => [
+            'bene-subnav-children',
+          ],
+        ],
+        '#items' => $page_child_links,
+      ];
+    }
+
+    return $build;
+  }
+
+  /**
+   * Gets an array of breadcrumb links up to (not including) the current page.
+   *
+   * @return array
+   *   Array of Link objects representing breadcrumbs.
+   */
+  private function getBreadcrumbs() {
+    /** @var \Drupal\Core\Routing\RouteMatch $current_route */
+    $current_route = \Drupal::service('current_route_match')->getCurrentRouteMatch();
```

Comment:
Preference for using services like this rather than using dependency injection? Works either way but the preferred Drual 8 approach is to use dependency injection when possible.